pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -41,9 +45,7 @@ const (
 	// use finalizers to manage the cleanup.
 	FinalizerName = "hyperconvergeds.hco.kubevirt.io"
 
-	// Foreground deletion finalizer is blocking removal of HyperConverged until explicitly dropped.
-	// TODO: Research whether there is a better way.
-	foregroundDeletionFinalizer = "foregroundDeletion"
+	HyperConvergedName = "hyperconverged-cluster"
```

Comment:
I think it will be better to let the user express a custom value with an environmental variable keeping this as a sane default.

Reply:
my reason for not pursuing this route is that it's going to be backed into the CSV already so the user will not be able to configure it when installing via OLM. Why not remove the variation and simply encode the value in the source?

Reply:
On the other side we are going to fail if, by mistake, the user creates the CR with a different name.
We can set consume this a default, set an override via env variable in the CSV and compose the alm-example in the CSV according to what we have in the environment variable.
See: https://github.com/kubevirt/hyperconverged-cluster-operator/pull/399

Reply:
Why not just put the "hyperconverged-cluster" in the alm-example, done. How does using an environment variable solve the problem of

> On the other side we are going to fail if, by mistake, the user creates the CR with a different name.

This will still be a problem if we use an environment variable. The facts are, regardless of hardcoded in source or using an environment variable, that:

1. The user must create the resource in the same namespace as the operator
1. The name of that resource is determined long before the user "Subscribes" to the operator

This was the reason @dankenigsberg that I wanted to specifically handle this scenario where a user creates a HyperConverged resource that we weren't expecting, either:

1. They mistakenly created the resource with the wrong name or namespace. 
1. They mistakenly created the resource when one we are reconciling already exists.

In either case, the polite thing to do is notify the user (via status), "Hey, this CR doesn't look right."

Reply:
Found it awkward to touch wrongly created CR. But when thinking about it again I suspect it would be ok for the hyperconverged controller to track all hyperconverged CRs.

Anyway, this should be discussed on another PR.